powerspikegg/rawdata/fetcher/aggregator_test.py

```diff
@@ -122,6 +167,183 @@ def test_champion_and_league_filtering(self):
+            "pentaKills": 0,
+            "physicalDamageDealt": 23168,
+            "physicalDamageDealtToChampions": 5802,
+            "physicalDamageTaken": 4752,
+            "quadraKills": 0,
+            "sightWardsBoughtInGame": 0,
+            "totalDamageDealt": 79151,
+            "totalDamageDealtToChampions": 12667,
+            "totalHeal": 462,
+            "totalTimeCrowdControlDealt": 489,
+            "totalUnitsHealed": 1,
+            "towerKills": 0,
+            "tripleKills": 0,
+            "trueDamageDealt": 40,
+            "trueDamageDealtToChampions": 0,
+            "trueDamageTaken": 0,
+            "visionWardsBoughtInGame": 2,
+            "wardsKilled": 2,
+            "wardsPlaced": 7,
+            "totalScoreRank": 0,
+            "totalDamageTaken": 10852,
+        }
+        result = aggregator.AverageStatisticsOnQuery(self.collection, query)
+        self.assertDictEqualWithDebug(result, expected)
+
+    def test_avg_champion_filtering(self):
+        """Tests filter per champion on the average aggregation."""
+        # 1 occurence in sample 1
+        query = service_pb2.Query(champion=constants_pb2.Champion(id=55))
+
+        expected = {
+            "total": 1,
+            "assists": 3,
+            "champLevel": 12,
+            "deaths": 2,
+            "doubleKills": 2,
+            "goldEarned": 8205,
+            "goldSpent": 7300,
+            "inhibitorKills": 0,
+            "killingSprees": 0,
+            "kills": 3,
+            "largestCriticalStrike": 0,
+            "largestKillingSpree": 0,
+            "largestMultiKill": 1,
+            "magicDamageDealt": 55942,
+            "magicDamageDealtToChampions": 6865,
+            "magicDamageTaken": 6099,
+            "minionsKilled": 165,
+            "neutralMinionsKilled": 0,
+            "neutralMinionsKilledEnemyJungle": 0,
+            "neutralMinionsKilledTeamJungle": 0,
+            "pentaKills": 0,
+            "physicalDamageDealt": 23168,
+            "physicalDamageDealtToChampions": 5802,
+            "physicalDamageTaken": 4752,
+            "quadraKills": 0,
+            "sightWardsBoughtInGame": 0,
+            "totalDamageDealt": 79151,
+            "totalDamageDealtToChampions": 12667,
+            "totalHeal": 462,
+            "totalTimeCrowdControlDealt": 489,
+            "totalUnitsHealed": 1,
+            "towerKills": 0,
+            "tripleKills": 0,
+            "trueDamageDealt": 40,
+            "trueDamageDealtToChampions": 0,
+            "trueDamageTaken": 0,
+            "visionWardsBoughtInGame": 2,
+            "wardsKilled": 2,
+            "wardsPlaced": 7,
+            "totalScoreRank": 0,
+            "totalDamageTaken": 10852,
+        }
+        result = aggregator.AverageStatisticsOnQuery(self.collection, query)
+        self.assertDictEqualWithDebug(result, expected)
+
+    def test_avg_league_and_champions_filtering(self):
+        """Tests filter per champion on the average aggregation."""
```

Comment:
Also, every "... played are selectionned." line needs to be re-written (I can't comment on these as it is not on this PR).